CondTools/RunInfo/src/LHCInfoPerFillPopConSourceHandler.cc

```diff
@@ -68,36 +70,54 @@ namespace cond {
 }  // namespace cond
 
 LHCInfoPerFillPopConSourceHandler::LHCInfoPerFillPopConSourceHandler(edm::ParameterSet const& pset)
-    : m_debug(pset.getUntrackedParameter<bool>("debug", false)),
+    : m_debug(pset.getUntrackedParameter<bool>("debug")),
       m_startTime(),
       m_endTime(),
-      m_samplingInterval((unsigned int)pset.getUntrackedParameter<unsigned int>("samplingInterval", 300)),
-      m_endFillMode(pset.getUntrackedParameter<bool>("endFill", true)),
-      m_name(pset.getUntrackedParameter<std::string>("name", "LHCInfoPerFillPopConSourceHandler")),
-      m_connectionString(pset.getUntrackedParameter<std::string>("connectionString", "")),
-      m_ecalConnectionString(pset.getUntrackedParameter<std::string>("ecalConnectionString", "")),
-      m_dipSchema(pset.getUntrackedParameter<std::string>("DIPSchema", "")),
-      m_authpath(pset.getUntrackedParameter<std::string>("authenticationPath", "")),
-      m_omsBaseUrl(pset.getUntrackedParameter<std::string>("omsBaseUrl", "")),
+      m_samplingInterval((unsigned int)pset.getUntrackedParameter<unsigned int>("samplingInterval")),
+      m_endFillMode(pset.getUntrackedParameter<bool>("endFill")),
+      m_name(pset.getUntrackedParameter<std::string>("name")),
+      m_connectionString(pset.getUntrackedParameter<std::string>("connectionString")),
+      m_ecalConnectionString(pset.getUntrackedParameter<std::string>("ecalConnectionString")),
+      m_dipSchema(pset.getUntrackedParameter<std::string>("DIPSchema")),
+      m_authpath(pset.getUntrackedParameter<std::string>("authenticationPath")),
+      m_omsBaseUrl(pset.getUntrackedParameter<std::string>("omsBaseUrl")),
       m_fillPayload(),
       m_prevPayload(),
       m_tmpBuffer() {
-  if (pset.exists("startTime")) {
+  if (pset.existsAs<std::string>("startTime", false)) {
```

Comment:
you dont need the `existsAs` since you have a fillDesc (and anyway `existsAs` is discouraged)